### PR TITLE
Integrate the beta clippy in CI

### PR DESCRIPTION
**Pull request by duguorong009**
Reference: #855

### CODE REVIEW

**Review by mrmr1993 — APPROVED** [review]
Nice, this is very cool!

**Comment by duguorong009** [comment]
Thanks! @mrmr1993 
One question: can you go for merge or the PR needs extra approval of @mimoo?

**Comment by mrmr1993** [comment]
> can you go for merge

Yes, once CI has passed I'm happy to merge, unless you also wanted @mimoo to review?

**Pull request merged**
